src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ProcessedEnvelopeNotificationHandler.java

```diff
@@ -0,0 +1,158 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.tasks;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.servicebus.ExceptionPhase;
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.IMessageHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.EnvelopeNotFoundException;
+import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.InvalidMessageException;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.in.msg.ProcessedEnvelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.EnvelopeFinaliserService;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.servicebus.MessageAutoCompletor;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+
+public class ProcessedEnvelopeNotificationHandler implements IMessageHandler {
+
+    private static final Logger log = LoggerFactory.getLogger(ProcessedEnvelopeNotificationHandler.class);
+
+    private final EnvelopeFinaliserService envelopeFinaliserService;
+    private final ObjectMapper objectMapper;
+    private final MessageAutoCompletor messageCompletor;
+
+    public ProcessedEnvelopeNotificationHandler(
+        EnvelopeFinaliserService envelopeFinaliserService,
+        ObjectMapper objectMapper,
+        MessageAutoCompletor messageCompletor
+    ) {
+        this.envelopeFinaliserService = envelopeFinaliserService;
+        this.objectMapper = objectMapper;
+        this.messageCompletor = messageCompletor;
+    }
+
+    @Override
+    public CompletableFuture<Void> onMessageAsync(IMessage message) {
+        return CompletableFuture
+            .supplyAsync(() -> tryProcessMessage(message))
+            .thenAcceptAsync(processingResult -> tryFinaliseMessage(message, processingResult))
+            .handleAsync((v, error) -> {
+                if (error != null) {
+                    log.error(
+                        "An error occurred when trying to handle 'processed envelope' message with ID {}",
+                        message.getMessageId()
+                    );
+                }
+
+                return null;
```

Comment:
i don't think `return null` does the trick. i could be confusing with something else

Reply:
There's no instantiation of Void